src/util.cpp

```diff
@@ -418,6 +422,20 @@ void ReleaseDirectoryLocks()
     dir_locks.clear();
 }
 
+bool DirIsWritable(const fs::path& directory)
+{
+    // Use a UUID for a unique temp filename.
+    fs::path tmpFile = directory / boost::lexical_cast<std::string>((boost::uuids::random_generator())());
```

Comment:
no need for lexical cast and uuid here: you can use `fs::unique_path` if it needs to be unique, but I'm not sure if hardcoding a name for the probe file wouldn't be better for troubleshooting.

Another possibility would be to report a different failure from the LockDirectory function.

Reply:
Agree, just improve error handling.